ssg/build_remediations.py

```diff
@@ -656,17 +656,20 @@ def get_rule_dir_remediations(dir_path, remediation_type, product=None):
         return []
 
     results = []
+    common_results = []
     for remediation_file in sorted(os.listdir(remediations_dir)):
         file_name, file_ext = os.path.splitext(remediation_file)
         remediation_path = os.path.join(remediations_dir, remediation_file)
 
         if file_ext == ext and rules.applies_to_product(file_name, product):
             if file_name == 'shared':
-                results.append(remediation_path)
+                common_results.append(remediation_path)
+            elif file_name != product:
+                common_results.insert(0, remediation_path)
             else:
-                results.insert(0, remediation_path)
+                results.append(remediation_path)
```

Comment:
Another nitpick, `results` will contain the `remediation_path` for the very product specific remediation, right? i.e. "ubuntu2004".

In this case `results` doesn't strictly need to be a list, we don't expect to have multiple `bash/2004.sh`.

Reply:
Yeah I wasn't sure what I wanted to do here. In the future, if there were say, SCE content, you could envision having multiple entries for the same product (say, Perl, Python, and Bash content for `ubuntu2004`). With fancy CPE checks &c, you could maybe determine that the system (at runtime) doesn't have Perl or Python but does have Bash... so you'd want to take the first of the unique extension entries out of there. Maybe the SCE content would be in separate directories, I don't know. 

Since we're ultimately coercing the results to a list (and not performing the selection here), leaving `results` as a list allows for this possibility and allows same-typed addition later.

Otherwise, we'd use an assignment here but just coerce it to a list for the return eventually anyways (`[result] + common_results` or `common_results.insert(0, result)`).

I figured it was a wash. But happy to do whatever. 

Preferences?

Reply:
But SCE is about checking, right? Not remediating.

> Maybe the SCE content would be in separate directories, I don't know.

I think that would be the case....

> Since we're ultimately coercing the results to a list (and not performing the selection here),

Yes, but `results` aims to contain the top pick. I just found it strange that we would have multiple matches for top pick.
It was just a nitpick anyway.